devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -638,6 +648,49 @@ void prepare() throws Exception {
 
         }
 
+        private void setKotlinSpecificFlags(DevModeContext devModeContext) {
+            Plugin kotlinMavenPlugin = null;
+            for (Plugin plugin : project.getBuildPlugins()) {
+                if (plugin.getKey().equals(KOTLIN_MAVEN_PLUGIN_GA)) {
+                    kotlinMavenPlugin = plugin;
+                    break;
+                }
+            }
+
+            if (kotlinMavenPlugin == null) {
+                return;
+            }
+
+            getLog().debug("Kotlin Maven plugin detected");
+
+            // get the absolute path of the all-open plugin
+            String fullPath = null;
+            List<Dependency> dependencies = kotlinMavenPlugin.getDependencies();
+            for (Dependency dependency : dependencies) {
+                // TODO do we actually need this to resolve the artifact or can we use the repoSystem?
+                Artifact artifact = system.createDependencyArtifact(dependency);
```

Comment:
I would like if I didn't have to add `system` to `DevMojo` and just use `repoSystem`. Is that possible @aloubyansky ?

What I added here I pretty much copied from [here](https://github.com/JetBrains/kotlin/blob/v1.3.41/libraries/tools/kotlin-maven-plugin/src/main/java/org/jetbrains/kotlin/maven/KotlinCompileMojoBase.java#L294)

Reply:
Yes, sure, here is how https://github.com/quarkusio/quarkus/blob/master/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/MavenArtifactResolver.java#L197

Reply:
Thanks! I'll update the PR to use this

Reply:
Fixed